src/tooltip/tooltip.js

```diff
@@ -422,13 +423,67 @@ angular.module('ui.bootstrap.tooltip', ['ui.bootstrap.position', 'ui.bootstrap.s
               }
             }
 
+            // hide tooltips/popovers for clickaway trigger
+            function bodyHideTooltipBind(e) {
+              if (!ttScope.isOpen) {
+                return;
+              }
+              // get element that was clicked
+              var target_elem = angular.element(e.target);
+
+              // make sure the tooltip/popover link was not the element clicked (keep in mind, the click event on the link toggles show/hide already)
+              if (e.target != element[0]) {
+                // hide by default
+                var shouldHide = true;
+
+                // loop through all ttScopes in the stackedMap (openedTooltips)
+                var ottScopes = openedTooltips.keys();
+                for (var index in ottScopes) {
+                  var ottScope = ottScopes[index];
+
+                  // see if the target element's scope is ottScope or is a descendant of ottScope (meaning it's part of the tooltip/popover element)
+                  var curScope = target_elem.scope();
```

Comment:
@wesleycho You recommended that I use the stacked map to figure out if the user is clicking inside a tooltip/popover, instead of using the bootstrap classes for popover/tooltip. Is there a different way you were thinking of doing this? It appears to me that the stacked map is just a list of scopes and the element's close method. Can I add the element itself to the stacked map?